pkg/scalers/kafka_scaler.go

```diff
@@ -86,6 +101,28 @@ func parseKafkaMetadata(metadata map[string]string) (kafkaMetadata, error) {
 		meta.lagThreshold = t
 	}
 
+	meta.authMode = kafkaAuthModeForNone
+	if val, ok := metadata["authMode"]; ok {
+		mode := kafkaAuthMode(val)
+		if mode != kafkaAuthModeForNone && mode != kafkaAuthModeForSaslPlaintext && mode != kafkaAuthModeForSaslScramSha256 && mode != kafkaAuthModeForSaslScramSha512 {
+			return meta, fmt.Errorf("err auth mode %s given", mode)
+		}
+
+		meta.authMode = mode
+	}
+
+	if meta.authMode != kafkaAuthModeForNone {
+		if metadata["username"] == "" {
+			return meta, errors.New("no username given")
+		}
+		meta.username = metadata["username"]
+
+		if metadata["passwd"] == "" {
+			return meta, errors.New("no passwd given")
+		}
+		meta.passwd = metadata["passwd"]
```

Comment:
This should likely resolve from a secret / env similar to how things like `host` are handled in RabbitMQ so the user doesn’t have to supply and reference the password in plain text.

Reply:
hi @jeffhollan , done it.  supply and reference the username and password in secret. But I don't think it is a good idea to get authentication information through env, because a kafka cluster has many topics, and many applications may be consuming messages. It is not necessary to configure environment variables for each application.

Reply:
Makes sense - and for that [we have a `TriggerAuthentication` spec that may help here](https://keda.sh/concepts/authentication/#re-use-credentials-and-delegate-auth-with-triggerauthentication).  Realize it would be additional work to have the kafka scaler leverage that ( https://github.com/kedacore/keda/issues/338 ) so I'm fine to just go with this approach for now and can track #338 as a seperate PR.  Let me know your thoughts

Reply:
hi @jeffhollan ,  agree with what you said，TriggerAuthentication is a good idea. please review my the newest code，supply password, ca  by TriggerAuthentication. it support sasl_ssl auth.
thank you very much for your response!

Reply:
Awesome reviewing right now.  Thanks again @iyacontrol - appreciate it! 💯